README.md

```diff
@@ -1,4 +1,203 @@
+
+[hardhat]: https://hardhat.org/
+[hardhat-badge]: https://img.shields.io/badge/Built%20with-Hardhat-FFDB1C.svg
+[license]: https://opensource.org/licenses/AGPL-v3
+[license-badge]: https://img.shields.io/badge/License-AGPL_v3-blue.svg
+
+## Quickstart
+
+After [creating a new repository from this template](https://github.com/new?template_name=osx-plugin-template-hardhat&template_owner=aragon), cloning, and opening it in your IDE, run
+
+```sh
+yarn install && cd packages/contracts && yarn install && yarn build && yarn typechain
+```
+
+You can now develop a plugin by changing the `src/MyPlugin.sol` and `src/MyPluginSetup.sol` files. You can directly import contracts from [Aragon OSx](https://github.com/aragon/osx) as well as OpenZeppelin's [openzeppelin-contracts](https://github.com/OpenZeppelin/openzeppelin-contracts) and [openzeppelin-contracts-upgradeable](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable) that are already set up for you.
+
+```sol
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {IDAO, PluginUUPSUpgradeable} from "@aragon/osx/core/plugin/PluginUUPSUpgradeable.sol";
+import {SafeCastUpgradeable} from '@openzeppelin/contracts-upgradeable/utils/math/SafeCastUpgradeable.sol';
+
+contract MyPlugin is PluginUUPSUpgradeable {
+    //...
+};
+```
+
+The initial `MyPlugin` and `MyPluginSetup` example comes with unit test, integration test, and test helpers in the `package/contracts/test` folder that you can reuse.
+
+To build and test your contracts, run
+
+```sh
+yarn clean && yarn build && yarn test
+```
+
+## Project
+
+The root folder of the repo includes three subfolders:
+
+```markdown
+.
+├── packages/contracts
+│ ├── src
+│ ├── deploy
+│ ├── test
+│ ├── utils
+│ ├── ...
+│ └── package.json
+│
+├── packages/subgraph
+│ ├── src
+│ ├── scripts
+│ ├── manifest
+│ ├── tests
+│ ├── utils
+│ ├── ...
+│ └── package.json
+│
+├── packages/js-client
+│ ├── src
+│ ├── test
+│ ├── ...
+│ └── package.json
+│
+├── ...
+└── package.json
+```
+
+The root-level `package.json` file contains global `dev-dependencies` for formatting and linting. After installing them with
+
+```sh
+yarn install
+```
+
+you can run the associated `scripts`. With
+
+### Formatting
+
+```sh
+yarn prettier:check
+```
+
+all `.sol`, `.js`, `.ts`, `.json`, and `.yml` files will be format-checked according to the specifications in `.prettierrc` file.With
+
+```sh
+yarn prettier:write
+```
+
+the formatting is applied.
+
+### Linting
+
+With
+
+```sh
+yarn lint
+```
+
+`.sol`, `.js`, and `.ts` files in the subfolders are analyzed with `solhint` and `eslint`, respectively.
+
+## Contracts
+
+In `packages/contracts`, first run
+
+```sh
+yarn install
+```
+
+### Building
+
+First build the contracts and
+
+```sh
+yarn build
+```
+
+and generate the [typechain TypesCript bindings](https://github.com/dethcrypto/TypeChain) with
```

Comment:
```suggestion
and generate the [typechain TypesScript bindings](https://github.com/dethcrypto/TypeChain) with
```

Reply:
Fixed in 81d88ad94b115a0558ffd7895d45eb6951420467